File: algorithms/CPlusPlus/Recursion/fibonacci.cpp
#include<iostream>

/*
This code snippet utilizaes tree recursion to calculate
fibonacci number upto n terms. Time complexity O(2^n)
*/

int fib(int n);

int main() {

	int n{7};

	int r = fib(n);

	std::cout << r << "\n";
	return 0;
}

int fib(int n) {

	if (n <= 1) {
		return n;
	}

	else {
		return fib(n - 1) + fib(n - 2);
	}
}

File: algorithms/CSharp/src/Recursion/fibonacci.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Algorithms.Recursion {
    public class Fibonacci {

        public static int FibonacciUsingRecursion(int number) {

            if ( (number == 0) || (number == 1) ) {
                return number;
            }

            return FibonacciUsingRecursion(number - 1) + FibonacciUsingRecursion(number - 2);

        }

        public static void Main() {

            Console.WriteLine($"The Fibonacci of 10 is: {FibonacciUsingRecursion(10)}");

        }

    }
}
